test/SolidityParser.cpp

```diff
@@ -703,6 +703,48 @@ BOOST_AUTO_TEST_CASE(literal_constants_with_ether_subdenominations_in_expression
 	BOOST_CHECK_NO_THROW(parseTextExplainError(text));
 }
 
+BOOST_AUTO_TEST_CASE(enum_declaration)
+{
+	char const* text = R"(
+		contract c {
+			enum foo { WARNING, NOTICE, ERROR, CRITICAL };
+			function c ()
+			{
+				a = foo.CRITICAL;
+			}
+			uint256 a;
+		})";
+	BOOST_CHECK_NO_THROW(parseTextExplainError(text));
+}
+
+BOOST_AUTO_TEST_CASE(empty_enum_declaration)
+{
+	char const* text = R"(
+		contract c {
+			enum foo { };
+			function c ()
```

Comment:
Also here (and below): The test should test exactly one thing (unless this makes it really hard to write).